management_api_app/api/routes/workspaces.py

```diff
@@ -1,24 +1,35 @@
-from fastapi import APIRouter, Depends
+from fastapi import APIRouter, Depends, HTTPException
+from starlette import status
 
 from api.dependencies.workspaces import get_repository, get_workspace_by_workspace_id_from_path
+from db.errors import UnableToAccessDatabase
 from db.repositories.workspaces import WorkspaceRepository
 from models.domain.resource import Resource
-from models.schemas.resource import ResourcesInList, ResourceInResponse
+from models.schemas.resource import ResourcesInList, ResourceInResponse, ResourceInCreate
+from resources import strings
 
 
 router = APIRouter()
 
 
-@router.get("/workspaces", response_model=ResourcesInList, name="workspaces:get-active-workspaces")
-async def retrieve_active_workspaces(
-        workspace_repo: WorkspaceRepository = Depends(get_repository(WorkspaceRepository))
-) -> ResourcesInList:
-    workspaces = workspace_repo.get_all_active_workspaces()
-    return ResourcesInList(resources=workspaces)
+@router.get("/workspaces", response_model=ResourcesInList, name=strings.API_GET_ALL_WORKSPACES)
```

Comment:
instead of doing error checking here you should use a middleware and fail from there if state store is down. Then you can remove all error handling from here and make code simpler and remove duplication

Reply:
@sachinkundu Can you provide a suggestion of how we could do this.

Reply:
``
@app.middleware("http")
async def add_check_statestore(request: Request, call_next):
    try:
       check_state_store()
   except:
       503
    return await call_next(request)
``

From https://fastapi.tiangolo.com/tutorial/middleware/

Reply:
Looks neat - will add this...

Reply:
Created #145